### DISCUSSION TITLE
Setting $indexOptions ['skipCreateModal']  to true seems to do nothing

### DISCUSSION THREAD

**Discussion by gglerum** [area17/twill#2451] [Q&A] [answered]
## Description

I don't want to show the modal on creating a new item.  On pressing the "new" button the form is opened and the modal is displayed on top.

### Steps to reproduce

- I've set "skipCreateModal" to true in the $indexOptions of the Controller.
- Clicked on the new button above the table

### Expected result

The form opens and I see an empty form without the modal on overlaid.

### Actual result

The  modal is shown overlaid on the form.

### Current solution that I found ##
```
    /**
     * Form settings
     *
     * @param [type] $request
     * @return void
     */
    protected function formData($request)
    {
        return [
            'createWithoutModal' => false
        ]
    }
```
The above code makes sure the create modal is not shown on pressing the new button. Which is weird, you'd expect "createWithoutModal" would have to be set to "true" for this to work.

### Versions

area17/twill ^2.0
laravel/framework ^9.19
PHP 8.1.0 
InnoDB ROW_FORMAT=DYNAMIC

**Comment by ifox** [comment, marked as answer]
I understand the confusion and we could certainly come up with a better name, but to clarify, the modal is called the "create" modal. In the index options, you can skip the create modal to go to the form directly. It still opens the modal on top of the form, so that you can set a title before continuing to fill up the form before creating the record. If you want to land on the form and be able to create the record without the modal step, which really is an edge case since everything in Twill is supposed to have at least one column for naming the record, title by default, which is configurables. 

To be clear, when you are not skipping the modal on the index, the record is created first, and then the full form saves are updates. Those features were added for developers that didn't want to use nullable columns and enforce users to fill up all fields before creating a record.

**Reply by gglerum** [reply]
Thanks for the answer!
